include/torch-mlir/Dialect/Torch/IR/TorchOps.td

```diff
@@ -1337,4 +1338,67 @@ def Torch_DtypeCalculateYieldDtypesOp : Torch_Op<"dtype.calculate.yield.dtypes",
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Symbolic shape modeling ops for TorchDynamo frontend.
+//===----------------------------------------------------------------------===//
+
+def Torch_SymbolicIntOp : Torch_Op<"symbolic_int", [Pure]> {
```

Comment:
Personally, I find 'symbolic_int' to be an odd name. Does that mean there are other symbolic types envisioned - a float perhaps ? Or a non-symbolic int ? 

If we reasonably assume dim shapes are only ever going to be constrained i64 ranges, would something like symbolic_dim,  dymbolic_shape or symbolic_dim_shape be better? A name that fits more closely with the realted op bind_symbolic_shape would be ideal.

Reply:
I think Torch refers to these everywhere as SymInt (there are also SymFloat and SymBool). Not strictly what is being referenced here. This is more of a torch.sym_int_constant for SymInts that exist purely within the program, but I'm ok leaving it how it is (i.e. it makes sense coming from the Torch view of things).

Reply:
Yeah precisely ⬆️ , didn't want to call it `symbolic_dim` to allow for future expansions where we might support importing `symbolic_float` and `symbolic_bool`s as well. Right now we insert these strictly for the SymInt case.